tests/components/ecobee/test_number.py

```diff
@@ -0,0 +1,117 @@
+"""The test for the ecobee thermostat number module."""
+from datetime import timedelta
+from unittest.mock import patch
+
+from pytest import MonkeyPatch
+
+from homeassistant.components.ecobee.const import DOMAIN as ECOBEE_DOMAIN
+from homeassistant.components.number import ATTR_VALUE, DOMAIN, SERVICE_SET_VALUE
+from homeassistant.const import ATTR_ENTITY_ID, UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.util import dt
+
+from .common import setup_platform
+
+from tests.common import async_fire_time_changed
+
+VENTILATOR_MIN_HOME_ID = "number.ecobee_ventilator_min_time_home"
+VENTILATOR_MIN_AWAY_ID = "number.ecobee_ventilator_min_time_away"
+THERMOSTAT_ID = 0
+
+
+async def test_ventilator_min_on_home_attributes(hass):
+    """Test the ventilator number on home attributes are correct."""
+    await setup_platform(hass, DOMAIN)
+
+    state = hass.states.get(VENTILATOR_MIN_HOME_ID)
+    assert state.state == "20"
+    assert state.attributes.get("min") == 0
+    assert state.attributes.get("max") == 60
+    assert state.attributes.get("step") == 5
+    assert state.attributes.get("friendly_name") == "ecobee Ventilator min time home"
+    assert state.attributes.get("unit_of_measurement") == UnitOfTime.MINUTES
+
+
+async def test_ventilator_min_on_away_attributes(hass):
+    """Test the ventilator number on away attributes are correct."""
+    await setup_platform(hass, DOMAIN)
+
+    state = hass.states.get(VENTILATOR_MIN_AWAY_ID)
+    assert state.state == "10"
+    assert state.attributes.get("min") == 0
+    assert state.attributes.get("max") == 60
+    assert state.attributes.get("step") == 5
+    assert state.attributes.get("friendly_name") == "ecobee Ventilator min time away"
+    assert state.attributes.get("unit_of_measurement") == UnitOfTime.MINUTES
+
+
+async def test_set_min_time_home(hass: HomeAssistant, monkeypatch: MonkeyPatch):
+    """Test the number can set min time home."""
+    target_value = 40
+    with patch(
+        "homeassistant.components.ecobee.Ecobee.set_ventilator_min_on_time_home"
+    ) as mock_set_min_home_time:
+        await setup_platform(hass, DOMAIN)
+
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_SET_VALUE,
+            {ATTR_ENTITY_ID: VENTILATOR_MIN_HOME_ID, ATTR_VALUE: target_value},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        mock_set_min_home_time.assert_called_once_with(THERMOSTAT_ID, target_value)
+
+    data = hass.data[ECOBEE_DOMAIN]
```

Comment:
Not sure how to do this one tbh

Reply:
https://github.com/home-assistant/core/pull/86100#discussion_r1083653828

Reply:
thanks

Reply:
Still looking, actually do we really need the second part of those test, 

```python
data = hass.data[ECOBEE_DOMAIN]

    monkeypatch.setitem(
        data.ecobee.get_thermostat(THERMOSTAT_ID)["settings"],
        "ventilatorMinOnTimeHome",
        target_value,
    )

    with patch("homeassistant.components.ecobee.Ecobee.update"):
        async_fire_time_changed(
            hass,
            dt.utcnow() + timedelta(minutes=5),
        )
        await hass.async_block_till_done()

    state1 = hass.states.get(VENTILATOR_MIN_HOME_ID)
    assert state1.state == str(target_value)
```

Looks like we are mocking the value we are looking for, looks very manual. If integration stop publishing to this we would not catch it with this test.

Reply:
Issue I have with patching this part is I need to patch the dictionary for the response because the service itself is already tested. 

```python 
target_value = 0
    with patch(
        "pyecobee.Ecobee.set_ventilator_min_on_time_away"
    ) as mock_set_min_away_time:
        await setup_platform(hass, DOMAIN)

        await hass.services.async_call(
            DOMAIN,
            SERVICE_SET_VALUE,
            {ATTR_ENTITY_ID: VENTILATOR_MIN_AWAY_ID, ATTR_VALUE: target_value},
            blocking=True,
        )
        await hass.async_block_till_done()
        mock_set_min_away_time.assert_called_once_with(THERMOSTAT_ID, target_value)
```

Reply:
@bdraco what do you think about this ☝️

Reply:
That looks OK to me. I don't see any of the integration details being patched.

I'd probably change the `pyecobee.Ecobee.set_ventilator_min_on_time_away` patch target to be more exact (patch where its imported if possible), or store the patched object and modify it later instead